modules/hal_nordic/nrfx/CMakeLists.txt

```
@@ -128,8 +128,19 @@ if(CONFIG_NRFX_TWI OR CONFIG_NRFX_TWIM)
   zephyr_library_sources(${SRC_DIR}/nrfx_twi_twim.c)
 endif()
 
-if (CONFIG_NRF_GRTC_TIMER AND CONFIG_NRF_GRTC_TIMER_CLOCK_MANAGEMENT)
-  zephyr_compile_definitions(NRF_GRTC_HAS_EXTENDED=1)
+if (CONFIG_NRF_GRTC_TIMER)
+	if (CONFIG_NRF_GRTC_TIMER_CLOCK_MANAGEMENT)
+	  zephyr_compile_definitions(NRF_GRTC_HAS_EXTENDED=1)
+	endif()
```

Nit, applies to all configs below:
```suggestion
	  zephyr_compile_definitions_ifdef(CONFIG_NRF_GRTC_TIMER_CLOCK_MANAGEMENT NRF_GRTC_HAS_EXTENDED=1)
```